include/swift/AST/PrintOptions.h

```diff
@@ -294,6 +275,9 @@ struct PrintOptions {
   /// Whether to print the extensions from conforming protocols.
   bool PrintExtensionFromConformingProtocols = false;
 
+  std::shared_ptr<ShouldPrintChecker> CurrentPrintabilityChecker =
```

Comment:
Like `Optional<ShouldPrintChecker*>`? does it mean we don't release the original one when when resetting the pointer?